src/color/p5.Color.js

```diff
@@ -53,8 +53,8 @@ p5.Color = function(renderer, vals) {
 
 p5.Color.prototype.toString = function() {
   var a = this.levels;
-  a[3] = this._array[3];  // String representation uses normalized alpha.
-  return 'rgba('+a[0]+','+a[1]+','+a[2]+','+ a[3] +')';
+  var alpha = this._array[3];  // String representation uses normalized alpha.
+  return 'rgba('+a[0]+','+a[1]+','+a[2]+','+ alpha +')';
```

Comment:
This is an awesome catch, thanks for the PR! I've added some unit tests for this behavior, feel free to pull them in from #1463 if you'd like otherwise the two PRs can merge in as siblings. This implementation looks good, 👍 !